File: leetcode/C++/score-of-parentheses.cpp
class Solution {
public:
    int scoreOfParentheses(string s) {
        vector<int> st = {0};
        for(int i=0,val=0; i<s.length(); ++i){
            if(s[i]=='(') st.push_back(0);
            else{
                val=st.back(); st.pop_back();
                st.back() += (val>0)? 2*val : 1;
            }
        }
        return st.back();
    }
};

File: leetcode/Python/score-of-parentheses.py
class Solution:
    def scoreOfParentheses(self, s: str) -> int:
        st, val = [0], 0
        for ch in s:
            if ch == '(':
                st.append(0)
            else:
                val = st.pop()
                st[-1] += 2*val if val > 0 else 1
        return st[-1]